packages/core/src/App/Containers/Routes/routes.jsx

```diff
@@ -58,14 +58,20 @@ const Routes = ({
 
     const lang = getLanguage();
     const lang_regex = /[?&]lang=/;
+    const has_lang = lang_regex.test(location.search);
 
     if (has_error) {
         return <Error {...error} />;
     }
 
     // we need to replace state of history object on every route
-    // to prevent language query parameter from disappering for non-english languages
-    if (!lang_regex.test(location.search) && lang !== 'EN') {
+    // to prevent language query parameter from disappering
+    // for non-english languages. Upon visiting with a
+    // non-supported language, the language still
+    // shows up in the URL. This is not in sync
+    // with the default language (EN), so we
+    // will remove it.
+    if ((!has_lang && lang !== 'EN') || (has_lang && lang === 'EN')) {
```

Comment:
I'm a bit confused regarding the second condition. Are you trying to keep `?lang=EN` if it's already a part of url?

Reply:
The first condition keeps the &lang={any country} in the URL when a supported translation is being used, the second condition removes the &lang= from the link ehen switching to the english language. The &lang= by default is removed from the link when using EN. If I don't do this, for example a user can have &lang=PL in the URL, even though he/she is using the english translation. This usually happens when for example the user registers an account while having polish language active, then the user gets a URL with the &lang=PL to activate the account.

Reply:
The second condition looks like it checks if url has `&lang=EN` and it will keep it inside the url. Ii seems you are trying to achieve the opposite behavior?

Reply:
`has_lang` only checks if there is &lang= part is in the url. `lang` checks the currently set language that is set in the localstore, this is not directly related to the url.

So basically, if the language is set to english, but it also has for example &lang=RU in the url, it will remove that part by setting the urlForLanguage to english. urlForLanguage('EN') by default doesn't have &lang= in the url.